### DISCUSSION TITLE
Partitioned Storage Deprecation Trial works unexpectedly

### DISCUSSION THREAD

**Discussion by rasulf** [privacysandbox/privacy-sandbox-dev-support#218] [General]
I have top level site a.com that embeds b.com iframe. I am using Broadcast Channel API to allow communication between the two - specifically - b.com must be able to write into the (now partitioned) localStorage of a.com.

I participated in the Partitioned Storage Deprecation Trial and generated a deprecation token with origin as b.com and included it in a meta tag in the head of my a.com document, before the b.com iframe loads. And everything works as expected - a.com localStorage is "unpartitioned" now.

I wanted to do the same thing for another c.com iframe that a.com embeds. While testing I saw that if I created a Deprecation Trial Token with a.com as origin - it allows c.com to access a.com's localStorage. This is unexpected. I never really allowed c.com to specifically allow access to a.com's storage. Why is my storage unpartitioned over here?

**Comment by rasulf** [comment]
Putting this under Issues.

**Comment by arichiv** [comment]
Answering in https://github.com/miketaylr/partitioned-storage-deprecation-trial-feedback/issues/11#issuecomment-1874421202
